src/sdl/SDL.cpp

```diff
@@ -346,6 +352,32 @@ char* sdlGetFilename(char* name)
     return filebuffer;
 }
 
+char* sdlGetFilePath(char* name)
+{
+    static char filebuffer[2048];
+
+    int len = strlen(name);
+
+    char* p = name + len - 1;
+
+    while (true) {
+        if (*p == FILE_SEP) {
+            break;
+        }
+        len--;
+        p--;
+        if (len == 0)
+            break;
+    }
```

Comment:
This is the sort of horrible C-style string code I'd prefer to work on removing not adding more of.

Could you use something like this instead:

https://stackoverflow.com/a/8518793/262458

Reply:
I think I got it. I have an idea using only C, though. Check if you like it.